spec/lib/salus/report_spec.rb

```diff
@@ -349,4 +349,230 @@ def build_report(report_uri)
+      directives = [
+        { 'uri' => http_url_one, 'format' => 'json' },
+        { 'uri' => file_path, 'format' => 'json' },
+        { 'uri' => http_url_two, 'format' => 'json' }
+      ]
+      report = build_report(
+        directives,
+        'uri:http*'
+      )
+
+      stub_request(:post, http_url_one)
+        .with(headers: { 'Content-Type' => 'application/json' }, body: report.to_json)
+        .to_return(status: 202)
+      stub_request(:post, http_url_two)
+        .with(headers: { 'Content-Type' => 'application/json' }, body: report.to_json)
+        .to_return(status: 202)
+
+      report.export_report
+
+      assert_requested(
+        :post,
+        http_url_one,
+        headers: { 'Content-Type' => 'application/json' },
+        body: report.to_json,
+        times: 1
+      )
+      assert_requested(
+        :post,
+        http_url_two,
+        headers: { 'Content-Type' => 'application/json' },
+        body: report.to_json,
+        times: 1
+      )
+      expect(File.exist?(file_path)).to eq(false)
+    end
+
+    it 'runs only `good-name` reports when `name:good-name` filter is provided' do
+      http_url_one = 'https://nerv.tk3/salus-report'
+      http_url_two = 'https://nerv.tk4/salus-report2'
+      file_path = './spec/fixtures/report/salus_report.json'
+      directives = [
+        { 'uri' => http_url_one, 'format' => 'json', 'name' => 'good-name' },
+        { 'uri' => file_path, 'format' => 'json', 'name' => 'good-name' },
+        { 'uri' => http_url_two, 'format' => 'json', 'name' => 'bad-name' }
+      ]
+      report = build_report(
+        directives,
+        'name:good-name'
+      )
+
+      stub_request(:post, http_url_one)
+        .with(headers: { 'Content-Type' => 'application/json' }, body: report.to_json)
+        .to_return(status: 202)
+      stub_request(:post, http_url_two)
+        .with(headers: { 'Content-Type' => 'application/json' }, body: report.to_json)
+        .to_return(status: 202)
+
+      report.export_report
+
+      assert_requested(
+        :post,
+        http_url_one,
+        headers: { 'Content-Type' => 'application/json' },
+        body: report.to_json,
+        times: 1
+      )
+      assert_requested(
+        :post,
+        http_url_two,
+        headers: { 'Content-Type' => 'application/json' },
+        body: report.to_json,
+        times: 0
+      )
+      expect(File.exist?(file_path)).to eq(true)
+      remove_file(file_path)
+    end
+
+    it 'runs only yaml-formatted reports when `format:yaml` filter is provided' do
+      http_url_one = 'https://nerv.tk3/salus-report'
+      http_url_two = 'https://nerv.tk4/salus-report2'
+      file_path = './spec/fixtures/report/salus_report.json'
+      directives = [
+        { 'uri' => http_url_one, 'format' => 'yaml' },
+        { 'uri' => file_path, 'format' => 'json' },
+        { 'uri' => http_url_two, 'format' => 'yaml' }
+      ]
+      report = build_report(
+        directives,
+        'format:yaml'
```

Comment:
@joshua-ostrom The ticket you created says
```ruby
--reports type:sarif # run all sarif reports
```
So you were saying the keyword should be `type`, not `format`.
Was this a typo in the ticket?

Reply:
Good catch, yes format would be great!